src/public/ui.h

```diff
@@ -18,6 +18,18 @@
 #include "public/build_config.h"
 #include "public/tools.h"
 
+#ifdef _WIN32
+#include "win32port.h"
+#else
+#include <stdint.h>
+#endif
+
+#ifdef _MSC_VER
+#define NORETURN __declspec(noreturn)
+#else
+#define NORETURN __attribute__((noreturn))
```

Comment:
This'll probably need to be it's own header if someone adds a check for C++11's attribute syntax